src/main/java/com/kalessil/phpStorm/phpInspectionsEA/inspectors/codeSmell/NamingConvention/NamingRule.java

```diff
@@ -0,0 +1,65 @@
+package com.kalessil.phpStorm.phpInspectionsEA.inspectors.codeSmell.NamingConvention;
+
+
+import com.jetbrains.php.lang.psi.elements.PhpClass;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+
+public class NamingRule {
+
+    final public static String TYPE_INTERFACE = "interface";
+    final public static String TYPE_TRAIT = "trait";
+    final public static String TYPE_FINAL = "final";
+    final public static String TYPE_ABSTRACT = "abstract";
+    final public static String TYPE_CLASS = "class";
+
+
+    final private String nameRegex;
+    final private String type;
+
+    @Nullable
+    private final String extendFqn;
+
+    public NamingRule(String nameRegex, String type, @Nullable String extendFqn) {
```

Comment:
With the idea does very valuable NPE (null pointer exception)-related analysis - must have in our case. As result the plugin is not throwing exceptions for long time already.